sympy/physics/control/lti.py

```diff
@@ -0,0 +1,1033 @@
+        >>> x0 = Matrix([y1, y2])
+        >>> ssm.evaluate(u, x0, t)
+        Matrix([
+        [Y1*exp(omega*t)],
+        [Y2*exp(omega*t)]])
+
+        One does not always have to integrate to obtain a solution. In this
+        case the do_integrals flag does nothing:
+
+        >>> ssm.evaluate(u, x0, t, do_integrals=False)
+        Matrix([
+        [Y1*exp(omega*t)],
+        [Y2*exp(omega*t)]])
+
+        But for more complicated examples evaluate retuns an Integral() object
+        when the flag is set:
+
+        >>> ssm = StateSpaceModel(Matrix([[-1, 1], [1, -1]]), ones(2, 1), eye(2))
+        >>> u = Matrix([exp(2 * t)])
+        >>> x0 = Matrix([1, 0])
+
+        >>> ssm.evaluate(u, x0, t)
+        Matrix([
+        [cosh(2*t)],
+        [sinh(2*t)]])
+
+        >>> ssm.evaluate(u, x0, t, do_integrals=False)
+        Matrix([
+        [Integral(exp(2*tau), (tau, 0, t)) + 1/2 + exp(-2*t)/2],
+        [Integral(exp(2*tau), (tau, 0, t)) + 1/2 - exp(-2*t)/2]])
+
+        >>> ssm.evaluate(u, x0, t, simplify=False)
+        Matrix([
+        [exp(2*t)/2 + exp(-2*t)/2],
+        [exp(2*t)/2 - exp(-2*t)/2]])
+
+            Numerical evaluation
+            ====================
+
+        If we give a tuple (t, t_list) instead of only the symbol, the system is
+        evaluated numericaly. The method returns a list of Matrices:
+
+        >>> ssm.evaluate(u, x0, (t, [0]))
+        [Matrix([
+        [1.0],
+        [  0]])]
+
+        >>> ssm.evaluate(u, x0, (t, [0, 0.2, 0.4, 0.6]))
+        [Matrix([
+        [1.0],
+        [  0]]), Matrix([
+        [ 1.08107237183845],
+        [0.410752325802816]]), Matrix([
+        [ 1.33743494630484],
+        [0.888105982187623]]), Matrix([
+        [1.81065556732437],
+        [1.50946135541217]])]
+
+        References
+        ==========
+
+        Joao P. Hespanha, Linear Systems Theory. 2009.
+        """
+        # verifiing valid arguments:
+        valid_flags = ('simplify', 'do_integrals')
+        for x in flags:
+            if x not in valid_flags:
+                raise ValueError('Unknown keyword argument: %s' % x)
+
+        try:
+            # assert right shape of u
+            if not u.shape[1] == 1:
+                raise ShapeError("u must not have more that one column, but has shape", u.shape)
+
+            if not self.represent[3].shape[1] == u.shape[0]:
+                raise ShapeError("u must have length", self.represent[3].shape[1])
+
+            # assert right shape of x0
+            if not x0.shape[1] == 1:
+                raise ShapeError("x0 must not have more than one column, but has shape", x0.shape)
+
+            if not self.represent[0].shape[1] == x0.shape[0]:
+                raise ShapeError("x0 must have length", self.represent[0].shape[1])
+
+        #
+        # Error handling
+        #
+        # if .shape goes wrong, a AttributeError is thrown
+        except AttributeError:
+            raise TypeError("u and x0 must be matrices!")
```

Comment:
This try block is too broad (too much code between try and except). AttributeError should not be caught anywhere. If you want to make a nice message for matrices then check at the start if they are matrices. Otherwise catching one exception to reraise another like this is a bad idea unless done very carefully.

Similar considerations apply to almost all use of try/except in this file.